### DISCUSSION TITLE
ImageLayer Tiles are loaded very slow when pitch is close to parallel to earth

### DISCUSSION THREAD

**Discussion by nebsar** [pelicanmapping/osgearth#1942] [Support Q&A] [answered]
When the view pitch is close to parallel to the earth, tiles of my image layer are rendered very slow and because of that I cannot see the proper tiles based on the view altitude. What could be the reason? How can I fix this? I am not using earth file. I am adding the layers programmatically. 

        GDALImageLayer* layerMbTiles = new GDALImageLayer;
	layerMbTiles->setProfile(Profile::create(Profile::GLOBAL_GEODETIC));
	layerMbTiles->setName("data layerl");
	layerMbTiles->setURL("data.mbtiles");
	layerMbTiles->setCachePolicy(osgEarth::CachePolicy::USAGE_READ_WRITE);
	mapNode->getMap()->addLayer(layerMbTiles);

I think if I use  below in and earth file, I think it works. But I would like to add the layers programmatically. 

![image](https://user-images.githubusercontent.com/5169314/151290981-8aa49d2b-7a6d-4345-8e97-05d63c6aecef.png)

![image](https://user-images.githubusercontent.com/5169314/151284280-a09af0de-783e-47ed-9787-2b6c41938664.png)

**Comment by pskpg** [comment]
This problem confused me for a long time too. 
By the way, layers without cache are rendered very slow.
By lowering the lod scale, rendering speed could be faster. But when a detailed image is needed, I have to wait for tens of seconds. 
Expect for solution!

**Reply by nebsar** [reply]
IMHO this needs to be addressed. I do not wan't make comparisons much, but even an opensource Java GIS Library NASA WorldWind, which does not implement modern OpenGL, provides faster tiling even when the view angle is close to parallel to the earth and even without cached tiles.
I really love osgEarth, but needs improvements so that more users implement more applications.
Any help is appreciated for this issue.
Thanks

**Reply by nebsar** [reply]
And I am not using WMS for map layer. MBTiles are on my harddrive.

**Comment by nebsar** [comment]
Ok. I noticed that, when I don't use Bing Imagery which cannot be cacheable, MBtiles are loaded fast enough. I found a Bing Imagery WMSlayer address that can be cached. I noticed that, when an internet source is added as layer, other layers' tile rendering is also affected negatively. But because of the cacheable WMSLayer the performance is better. Is it possible that multithreading does not work well during tile request and rendering? Is it possible that I use ImGui and ImGui decrease the performance?
Thanks

**Comment by gwaldron** [comment, marked as answer]
Each tile in osgEarth loads to completion before displaying. So if one layer is slow to load, it will slow down the whole tile. You can try to use the **experimental** async feature (`async="true"` on your layer) but just keep in mind it's experimental.

**Reply by nebsar** [reply]
Yeah I noticed the WMS layer prevents loading the local data layers faster. I will try that or I will get rid of the WMS layer and go on with the local data layers. Thank you Glenn.

Edit: I tried the async feature. I seems to work fine. :)
